scipy/stats/_covariance.py

```diff
@@ -147,8 +205,102 @@ def from_cholesky(L):
         because the Cholesky decomposition does not exist for a singular
         covariance matrix.
 
+        Examples
+        --------
+        Prepare a symmetric positive definite covariance matrix ``A`` and a
+        data point ``x``.
+
+        >>> import numpy as np
+        >>> from scipy import stats
+        >>> rng = np.random.default_rng()
+        >>> n = 5
+        >>> A = rng.random(size=(n, n))
+        >>> A = A @ A.T  # make the covariance symmetric positive definite
+        >>> x = rng.random(size=n)
+
+        Perform the Cholesky decomposition of ``A`` and create the
+        ``Covariance`` object.
+
+        >>> L = np.linalg.cholesky(A)
+        >>> cov = stats.Covariance.from_cholesky(L)
+
+        Compare the functionality of the ``Covariance`` object against
+        reference implementation.
+
+        >>> from scipy.linalg import solve_triangular
+        >>> res = cov.whiten(x)
+        >>> ref = solve_triangular(L, x, lower=True)
+        >>> np.testing.assert_allclose(res, ref)
+        >>> res = cov.log_pdet
+        >>> ref = np.linalg.slogdet(A)[-1]
+        >>> np.testing.assert_allclose(res, ref)
+
+        """
+        return CovViaCholesky(cholesky)
+
+    @staticmethod
+    def from_eigendecomposition(eigendecomposition):
+        r"""
+        Representation of a covariance provided via eigendecomposition
+
+        Parameters
+        ----------
+        eigendecomposition : sequence
+            A sequence (nominally a tuple) containing the eigenvalue and
+            eigenvector arrays as computed by `scipy.linalg.eigh` or
+            `numpy.linalg.eigh`.
```

Comment:
In `qmc.MultivariateNormalQMC` we are asking for the input to `eigh` instead. I am wondering if we should not do this instead. It would also mean having a similar interface with the other methods. WDYT?

Reply:
The input to `eigh` is the matrix itself, right? 
If they just pass in the matrix, then we have to try to figure out whether the matrix is singular or not, and if so, which of the eigenvalues are actually supposed to be treated as zero. (The choice fundamentally changes the distribution.) 
If they pass in the eigenvalues themselves, then they are in control of which are zero, and we just look out for eigenvalues that are exactly zero.

Reply:
Yes it would be the matrix itself. But ok since with Cholesky we also don't pass the matrix and ask users to have the decomposition on their side I suppose this is coherent.

Reply:
We discussed separately that we will leave this alone, but we will eventually have `scipy.stats.Covariance` accept the original matrix, e.g. `scipy.stats.Covariance(A)`. Under the hood it will do `cholesky` and return an object that behaves like a `CovViaCholesky` object. For singular matrices, users will be forced to choose a decomposition and ensure that zero values (e.g. eigenvalues) are exactly zero if they want them to be treated as zeros.